src/Threads.cpp

```diff
@@ -134,6 +131,9 @@ void CThreadPool::threadFunc() {
 	assert(m_num_threads > 0 && m_num_idle > 0);
 	m_num_threads--;
 	m_num_idle--;
+
+	if (m_num_threads == 0)
```

Comment:
m_done is not what we want here. m_done is true if destructor is called even if there are several threads are still running. The destructor wants to wait all threads are gone.

About m_idle, it is always "m_idle <= m_num_threads".